SimPEG/DCIP/BaseIP.py

```diff
@@ -0,0 +1,182 @@
+from SimPEG import *
+from BaseDC import SurveyDC, FieldsDC_CC
+
+class SurveyIP(SurveyDC):
+    """
+        **SurveyDC**
+
+        Geophysical DC resistivity data.
+
+    """
+
+    def __init__(self, srcList, **kwargs):
+        self.srcList = srcList
+        Survey.BaseSurvey.__init__(self, **kwargs)
+        self._Ps = {}
+
+    def dpred(self, m, u=None):
+        """
+            Predicted data.
+
+            .. math::
+                d_\\text{pred} = Pu(m)
+        """
+
+        return self.prob.forward(m)
+
+
+class ProblemIP(Problem.BaseProblem):
+    """
+        **ProblemIP**
+
+        Geophysical IP resistivity problem.
+
+    """
+
+    surveyPair = SurveyDC
+    Solver     = Solver
+    sigma = None
+    Ainv = None
+    u = None
+
+    def __init__(self, mesh, **kwargs):
+        Problem.BaseProblem.__init__(self, mesh)
+        self.mesh.setCellGradBC('neumann')
+        Utils.setKwargs(self, **kwargs)
+
+    # deleteTheseOnModelUpdate = ['_A', '_Msig', '_dMdsig']
+
+    @property
+    def Msig(self):
+        if getattr(self, '_Msig', None) is None:
```

Comment:
Since these are common to both DC and IP, should we have a `BaseDCIP` ? (this can be a thing we take care of after it has been merged onto dev). Also: out of curiosity, is there a reason for not using `mesh.getFaceInnerProduct`?